### PR TITLE
Add Zus System Test Framework

### PR DESCRIPTION
**Pull request by stewartie4**
Create a wrapper framework which delegates to the default go testing.T package.
Features:
- Individual test cases now have a default timeout of 20s for API, 40s for CLI
- This timeout can be overridden on a single test case basis, **which has flagged some very slow functionality/tests** (below)
- A panic in an individual test case no longer kills the entire test run in most cases (as far as go, in it's wisdom, allows 😆)
- Automatic logging of test case schedule, start and end with correctly formatted name rather than underscores for ease of search
- Tests now run in parallel by default, sequential tests are easy to spot now, and should be used sparingly  
- Ability to toggle off sequential or slow tests

Usage:
```
func TestFuncName(originalT *testing.T){
    t := test.NewSystemTest(originalT)
    t.Log("The wrapper can be used exactly like the standard *testing.T, which it delegates to, however new functions have also been added.")

    t.Run("Name", func(t *test.SystemTest) {
        ...
    })
    t.RunWithTimeout("Name", 60*time.Second, func(t *test.SystemTest) {
        ...
    })
    t.RunSequentially("Name", func(t *test.SystemTest) {
        ...
    })
    t.RunSequentiallyWithTimeout("Name", 60*time.Second, func(t *test.SystemTest) {
        delegateT := t.Unwrap
        delegateT.Log("t.Unwrap exposes the delegate *testing.T instance, bypassing the wrapper framework (not recommended).")
    })
}
```

Slow scenarios uncovered: (Anything over 2 minutes is serious enough to warrant a look)
image

Logging:
image

### CODE REVIEW

File: internal/api/util/test/system_test_framework.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
+package test
```
Comment (stewartie4) [inline internal/api/util/test/system_test_framework.go:1]: The crux is here

File: internal/api/util/test/system_test_framework.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+var DefaultTestTimeout = 20 * time.Second
+
+type SystemTest struct {
+	Unwrap       *testing.T
+	testComplete bool
+	isChildTest  bool
+}
+
+func NewSystemTest(t *testing.T) *SystemTest {
+	return &SystemTest{Unwrap: t, testComplete: false, isChildTest: false}
```
Comment (stewartie4) [inline internal/api/util/test/system_test_framework.go:19]: The crux of this PR is here

**Review by Kishan-Dhakan — DISMISSED** [review]
Minor comments, a much needed refactor!

File: tests/cli_tests/flaky___broken_scenarios_test.go [hunk 3]
```diff
@@ -31,11 +35,9 @@ func Test___FlakyBrokenScenarios(t *testing.T) {
 	t.Parallel()
 
 	// FIXME The test is failing due to sync function inability to detect the file changes in local folder see https://github.com/0chain/zboxcli/issues/250
-	t.Run("Sync path to non-empty allocation - locally updated files (in root) must be updated in allocation", func(t *testing.T) {
-		t.Parallel()
-
+	t.Run("Sync path to non-empty allocation - locally updated files (in root) must be updated in allocation", func(t *test.SystemTest) {
 		allocationID := setupAllocation(t, configPath, map[string]interface{}{"size": 2 * MB})
-		defer createAllocationTestTeardown(t, allocationID)
+		createAllocationTestTeardown(t, allocationID)
```
[the remark this answers is not in the document]
Comment (stewartie4) [reply tests/cli_tests/flaky___broken_scenarios_test.go:41]: nope, createAllocationTestTeardown now runs within t.Cleanup() so only runs after the test case has finished which is better than defer as it runs outside the test case timer

File: tests/cli_tests/zboxcli_cancel_allocation_test.go [hunk 4]
```diff
@@ -30,8 +32,12 @@ func TestCancelAllocation(t *testing.T) {
...
 	})
 
-	t.Run("No allocation param should fail", func(t *testing.T) {
-		t.Parallel()
+	t.Run("No allocation param should fail", func(t *test.SystemTest) {
+		dt := time.Now()
+		t.Logf("Test case [No allocation param should fail] before parallel at [%s] ", dt.Format("01-02-2006 15:04:05"))
+
+		dt = time.Now()
+		t.Logf("Test case [No allocation param should fail] after parallel at [%s] ", dt.Format("01-02-2006 15:04:05"))
```
[the remark this answers is not in the document]
Comment (stewartie4) [reply tests/cli_tests/zboxcli_cancel_allocation_test.go:40]: thanks this needs removed, was there for debug

File: tests/cli_tests/0_owner_update_config_test.go [hunk 5]
```diff
@@ -21,7 +25,7 @@ func TestOwnerUpdate(t *testing.T) {
 
 	newOwnerName := escapedTestName(t)
 
-	t.Run("should allow update of owner: StorageSC", func(t *testing.T) {
+	t.RunSequentiallyWithTimeout("should allow update of owner: StorageSC", 60*time.Second, func(t *test.SystemTest) {
```
[the remark this answers is not in the document]
Comment (stewartie4) [reply tests/cli_tests/0_owner_update_config_test.go:28]: This test consistently takes > 40s whereas the others never do.
Not sure why, but I would guess a performance issue in the storage smart contract 
image
I only want to add custom timeouts to test cases that I know for sure will time out

File: tests/cli_tests/zboxcli_cancel_allocation_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (stewartie4) [reply tests/cli_tests/zboxcli_cancel_allocation_test.go]: e4249d7

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**